pkg/apis/apps/types.go

```diff
@@ -44,6 +44,21 @@ type StatefulSet struct {
 	Status StatefulSetStatus
 }
 
+// PodCreationPolicyType defines the policy for creating pods under a stateful set.
+type PodCreationPolicyType string
+
+const (
+	// OrderedReadyPodCreation will create pods in strictly increasing order on
+	// scale up and strictly decreasing order on scale down, progressing only when
+	// the previous pod is ready or terminated. At most one pod will be changed
+	// at any time.
+	OrderedReadyPodCreation PodCreationPolicyType = "OrderedReady"
```

Comment:
+1 for "OrderedReady", once we have stable pods we can be backwards-compatible and add "OrderedStable".